common/utils.js

```diff
@@ -466,6 +466,11 @@
             return result;
         }
 
+        function isBrowserIE() {
+            //Internet Explorer 6-11
+            return /*@cc_on!@*/false || !!document.documentMode;
```

Comment:
From [here](http://stackoverflow.com/questions/10072816/how-does-this-ie-check-work):
"If IE sees this comment, it actually compiles the contents of comment blocks as pure code."

"IE compiles the ! as if it were written into source so it handles the code as if (!false), while all other browsers treat it as if (false)."

I was more curious about understanding why it works. `cc_on` suggests that this comment needs to be conditionally compiled so it writes the `!` into the compiled source.